src/commands/help.ts

```diff
@@ -68,18 +116,51 @@ export class HelpCommand extends Command {
       }
 
       const name = `${command.options.name}${command.options.guildOnly ? '(Guild only)' : ''}`
-      const addedLength = name.length + command.options.description.length
+      const added = `**${name}** - ${command.options.description}\n`
 
-      if (embed.fields.length === 25 || length + addedLength > 6000) {
+      if (length + added.length > 6000) {
         await msg.author.send({ embed })
         embed = new MessageEmbed(this.client).setTitle('Command list')
       }
 
-      embed.addField(name, command.options.description)
-      length += addedLength
+      if (content.length + added.length > 1024) {
+        embed.addField(groupName, content)
+
+        groupName = `${groupName} (cont.)`
+        length += groupName.length
+        content = ''
+      }
+
+      content += added
+      length += added.length
     }
 
-    await msg.author.send({ embed })
+    return { content, embed, length }
+  }
+
+  /**
+   * Maps the bot's commands by group.
+   */
+  private getCommandsByGroup(): Map<string, Command[]> {
+    const commands: Map<string, Command[]> = new Map()
+
+    for (const command of this.client.commands.values()) {
+      let group = command.options.group
+
+      if (!group) {
+        group = 'no group'
```

Comment:
This change introduces a magic string which we should try to avoid. I think to improve this, we could assign the command to a default group if the user doesnt specify it. This could be done when the command is first registered. I'll have to think about this one for a bit.